### PR TITLE
CRAYSAT-1837: Improve python-csm-api-client build

### PR DESCRIPTION
**Pull request by annapoorna-s-alt**
IM:CRAYSAT-1837
Reviewer:Ryan

python-csm-api-client builds were taking approximately 35 mins to complete the build.
Will improvise it to take less amount of time by adding poetry.lock file that resolves the dependencies

## Summary and Scope

_python-csm-api-client builds are taking 35 minutes_

To improvise the build to take less amount of time, we added the poetry.lock file which is an auto generated script from Poetry 1.5.1
poetry.lock file already has its transitive dependencies.
As we added poetry.lock file Jenkins does not need to resolve its dependencies, which takes a minimal amount of time.

## Issues and Related PRs

_Resolves [CRAYSAT-1837](https://jira-pro.it.hpe.com:8443/browse/CRAYSAT-1837)._

## Testing

_List the environments in which these changes were tested._

### Test description:

_Will observe the Jenkins build pipeline of `python-csm-api-client`._
_It takes only around 1m 36s to complete the build pipeline after adding poetry.lock file_

## Risks and Mitigations

_Minimal_

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by annapoorna-s-alt** [comment]
PR corresponds to the python-csm-api-client build issue: https://github.com/Cray-HPE/python-csm-api-client/pull/26

**Review by haasken-hpe — APPROVED** [review]
Strictly speaking, this should not be required since this only affects the build of the the python-csm-api-client repo on its own. However, it's also fine to make the update, so I'm approving.

Just remember to remove the modification to use the unstable repository in the `Dockerfile` after you merge https://github.com/Cray-HPE/python-csm-api-client/pull/26 and update the release/1.2 branch in that repo.

**Review by shivaprasad-metimath — APPROVED** [review]

**Comment by annapoorna-s-alt** [comment]
> Update the copyright line in the files changed with the year

Copyright is from the `dockerfile` changes to get unstable build. So we don't merge it, will stash that commit

**Pull request merged**
